x-pack/plugins/lens/public/indexpattern_datasource/operations/layer_helpers.ts

```diff
@@ -1506,3 +1557,225 @@ export function getManagedColumnsFrom(
+        layer: isLast
+          ? { indexPatternId: indexPattern.id, columns: {}, columnOrder: [] }
+          : computeLayerFromContext(metricsArray.length === 1, metricsArray, indexPattern),
+        columnId,
+        field: !metricContext?.isFullReference ? field ?? documentField : undefined,
+        columnParams: metricContext?.params ?? undefined,
+        incompleteFieldName: metricContext?.isFullReference ? field?.name : undefined,
+        incompleteFieldOperation: metricContext?.isFullReference
+          ? metricContext?.pipelineAggType
+          : undefined,
+        indexPattern,
+        visualizationGroups: [],
+      });
+      if (metricContext) {
+        metricContext.accessor = columnId;
+      }
+    }
+  }
+
+  // update the layer with the custom label and the format
+  let columnIdx = 0;
+  for (const [columnId, column] of Object.entries(layer.columns)) {
+    if (format) {
+      layer = updateColumnParam({
+        layer,
+        columnId,
+        paramName: 'format',
+        value: {
+          id: format,
+          params: {
+            decimals: 2,
+          },
+        },
+      });
+    }
+
+    // for percentiles I want to update all columns with the custom label
+    if (customLabel && column.operationType === 'percentile') {
+      layer = updateColumnLabel({
+        layer,
+        columnId,
+        customLabel,
+      });
+    } else if (customLabel && columnIdx === Object.keys(layer.columns).length - 1) {
+      layer = updateColumnLabel({
+        layer,
+        columnId,
+        customLabel,
+      });
+    }
+    columnIdx++;
+  }
+  return layer;
+}
+
+export function getSplitByTermsLayer(
+  indexPattern: IndexPattern,
+  splitField: IndexPatternField,
+  dateField: IndexPatternField | undefined,
+  layer: VisualizeEditorLayersContext
+): IndexPatternLayer {
+  const { termsParams, metrics, timeInterval } = layer;
+  const copyMetricsArray = [...metrics];
+  const computedLayer = computeLayerFromContext(
+    metrics.length === 1,
+    copyMetricsArray,
+    indexPattern,
+    layer.format,
+    layer.label
+  );
+
+  const columnId = generateId();
+  let termsLayer = insertNewColumn({
+    op: 'terms',
+    layer: insertNewColumn({
+      op: 'date_histogram',
+      layer: computedLayer,
+      columnId: generateId(),
+      field: dateField,
+      indexPattern,
+      visualizationGroups: [],
+      columnParams: {
+        interval: timeInterval,
+      },
+    }),
+    columnId,
+    field: splitField,
+    indexPattern,
+    visualizationGroups: [],
+  });
+  const termsColumnParams = termsParams as TermsIndexPatternColumn['params'];
+  if (termsColumnParams) {
+    for (const [param, value] of Object.entries(termsColumnParams)) {
+      let paramValue = value;
+      if (param === 'orderBy') {
+        const existingMetricColumn = Object.entries(termsLayer.columns)
+          .filter(([colId]) => isSortableByColumn(termsLayer, colId))
+          .map(([id]) => id)[0];
```

Comment:
nit: this operates only with `keys`, so one cycle can be skipped:

```suggestion
        const [existingMetricColumn] = Object.keys(termsLayer.columns)
          .filter((colId) => isSortableByColumn(termsLayer, colId));
```